### PR TITLE
Fix single port RRQ with options fail

### PR DESCRIPTION
**Pull request by kjyrinki-unikie**
When running tftpd with -s single port argument and Read Request with options are provided ACK for OACK is not properly received.

The problem is that code is waiting MPSC channel response, but there is no thread for receiving data from UDP socket and MPSC receive will timeout.

Fix is to move waiting of ACK response for OACK into worker thread in the beginning of file sending. I.e. File send is started with receiving ACK when there were options and OACK was sent in the accept_request().

Fixes issue #16

### CODE REVIEW

**Comment by altugbakan** [comment]
I think this is a good start, but can we solve this problem by creating a new method called `recv_raw`, which is just `recv` for the `UdpSocket`, but on `ServerSocket`, uses the internal `UdpSocket` to get the data? This way we can put `recv_raw` inside the `check_response` function, and it will use the provided `Socket` to get the data for each case.

**Comment by kjyrinki-unikie** [comment]
Not 100% sure if I got your idea, but I think approach you proposed might have issue in single port case if some other client tries to connect while the check_response() is waiting the ACK for OACK from the previous client in this new recv.

File: src/worker.rs [hunk 1]
```diff
@@ -69,13 +69,13 @@ impl<T: Socket + ?Sized> Worker<T> {
 
     /// Sends a file to the remote [`SocketAddr`] that has sent a read request using
     /// a random port, asynchronously.
-    pub fn send(self) -> Result<(), Box<dyn Error>> {
+    pub fn send(self, rcv_oack_ack: bool) -> Result<(), Box<dyn Error>> {
```
Comment (altugbakan) [inline src/worker.rs:72]: small nitpick: can it be `check_response` or `should_check_response` instead of `rcv_oack_ack`?

File: src/worker.rs [hunk 2]
```diff
@@ -130,10 +130,13 @@ impl<T: Socket + ?Sized> Worker<T> {
         Ok(())
     }
 
-    fn send_file(self, file: File) -> Result<(), Box<dyn Error>> {
+    fn send_file(self, file: File, rcv_oack_ack: bool) -> Result<(), Box<dyn Error>> {
```
Comment (altugbakan) [inline src/worker.rs:133]: Also here

**Review by altugbakan — COMMENTED** [review]
Yeah you're right, this solution is better. I have left some small comments, and if you can make the unit tests pass I'll go ahead and merge it.

**Review by altugbakan — APPROVED** [review]

**Pull request merged**